notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py

```diff
@@ -110,24 +111,31 @@
 # EPY: END code
 
 # EPY: START markdown
-# Use MERFISH-calculated size factors to scale the channels across the hybridization rounds and visualize the resulting filtered and scaled images
+# Use MERFISH-calculated size factors to scale the channels across the hybridization rounds and visualize the resulting filtered and scaled images. Right now we have to extract this information from the metadata and apply this transformation manually.
 # EPY: END markdown
 
 # EPY: START code
-stack_blurred = s.image.numpy_array
-sc = s.org['metadata']['scale']
-sc_df = pd.DataFrame([(int(k), v) for k,v in sc.items()], columns = ['bit', 'scale'])
-mp = pd.merge(s.squeeze_map, sc_df, on='bit', how='left')
-scale_dict = dict(zip(mp.ind.values, mp.scale.values))
+scale_factors = {(t['h'], t['c']): t['scale_factor'] for index, t in s.tile_metadata.iterrows()}
+# EPY: END code
+
+# EPY: START code
+# this is a scaling method. It would be great to use image.apply here. It's possible, but we need to expose H & C to 
+# at least we can do it with get_slice and set_slice right now.
 
-for k, v in scale_dict.items():
-    stack_blurred[k] = stack_blurred[k]/v
-    
-s.set_stack(s.un_squeeze(stack_blurred))
+for indices in s.image._iter_indices():
+    data = s.image.get_slice(indices)[0]
+    scaled = data / scale_factors[indices['h'], indices['c']]
+    s.image.set_slice(indices, scaled)
 # EPY: END code
 
 # EPY: START code
-tile_lims(stack_blurred, 2, size=10);
+from scipy.stats import scoreatpercentile
+# EPY: END code
+
+# EPY: START code
+mp = s.max_proj('h', 'c', 'z')
```

Comment:
Indices.HYB, Indices.CH, Indices.Z ?

Reply:
Good catch -- my tendency is to try to avoid our internals in notebooks, but this is important to keep them up to date. Will fix.